subsys/net/lib/http/Kconfig

```diff
@@ -37,6 +37,52 @@ config HTTP_SERVER
 	  HTTP server support.
 	  Note: this is a work-in-progress
 
+#if HTTP_SERVER
```

Comment:
Changes to this Kconfig file should have been made as part of [this commit](https://github.com/zephyrproject-rtos/zephyr/pull/59669/commits/caaf12d464cc59c8f9c7d15ffe2f25a948838375) instead of [this commit](https://github.com/zephyrproject-rtos/zephyr/pull/59669/commits/83587a3cbca9317d955763542b0a177b5202b7bd).

Can you please fix-up and reword the commit messages?